cunit/jmap_util.testc

```diff
@@ -189,6 +188,18 @@ static void test_decode_to_utf8(void)
         0.51,
         "Ad""\xef\xbf\xbd""la""\xef\xbf\xbd""de",
         0
+    }, {
+        // Multi-byte UTF-8 with invalid byte sequence:
+        // "Hello,😛🪐world🍎🏓!💾," "\xc3\x28"
+        "SGVsbG8s8J+Ym/CfqpB3b3JsZPCfjY7wn4+TIfCfkr4swyg=",
+        "utf-8",
+        ENCODING_BASE64,
+        0.0,
+        "\x48\x65\x6c\x6c\x6f\x2c\xf0\x9f\x98\x9b\xf0\x9f"
+        "\xaa\x90\x77\x6f\x72\x6c\x64\xf0\x9f\x8d\x8e\xf0"
+        "\x9f\x8f\x93\x21\xf0\x9f\x92\xbe"
+        "," "\xef\xbf\xbd" "(",
+        1
```

Comment:
I'm not familiar with these tests....  This chunk at least looks consistent with the neighbouring chunks, though I can't see whether it's good or bad by just looking at it, nor do I understand how it proves the solution.  Do you want me to dig deeper, or do you think it's enough that you understand it and know it's good?  If I was going to dig deeper, I think I would run this test suite under coverage, and see if the new code was covered.

Reply:
I think it is good enough as regression test as it is. Should the test ever fail we'll see a nicely formatted comparison of strings in a UTF-8 enabled terminal. I also ran the test on a Cyrus instance that does not have the fix applied to prove that it breaks there.